juju/charm.py

```diff
@@ -82,10 +87,8 @@ class CharmBase(Object):
 
     on = CharmEvents()
 
-    def __init__(self, framework, key, metadata, charm_dir):
-        super().__init__(framework, key)
-        self.metadata = metadata
-        self.charm_dir = charm_dir
+    def __init__(self, framework):
+        super().__init__(framework, None)
```

Comment:
We should probably keep the key, if only to make sure people realize there's some namespacing issue they must look after if they decide to create an instance of it.